Algorithms/Encoders/NYSIISEncoder.cs

```diff
@@ -0,0 +1,197 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace Algorithms.Encoders
+{
+    /// <summary>
+    /// Class for NYSIIS encoding strings.
+    /// </summary>
+    public class NYSIISEncoder
```

Comment:
Codacy found an issue: [Rename class 'NYSIISEncoder' to match pascal case naming rules, consider using 'NysiisEncoder'.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6311443)